extensions/reactive-pg-client/runtime/src/main/java/io/quarkus/reactive/pg/client/runtime/PgPoolTemplate.java

```diff
@@ -13,30 +13,23 @@
 @Template
 public class PgPoolTemplate {
 
-    // Visible for testing
-    static volatile PgPool pgPool;
-
     public RuntimeValue<PgPool> configurePgPool(RuntimeValue<Vertx> vertx, BeanContainer container,
             DataSourceConfig dataSourceConfig, PgPoolConfig pgPoolConfig, LaunchMode launchMode, ShutdownContext shutdown) {
 
-        initialize(vertx.getValue(), dataSourceConfig, pgPoolConfig);
+        PgPool pgPool = initialize(vertx.getValue(), dataSourceConfig, pgPoolConfig);
 
         PgPoolProducer producer = container.instance(PgPoolProducer.class);
         producer.initialize(pgPool);
 
         if (!launchMode.isDevOrTest()) {
```

Comment:
It might be a stupid question but why don't you close the pool when testing or in dev mode?